go/arrow/datatype_nested.go

```diff
@@ -148,6 +148,40 @@ func (t *StructType) FieldByName(name string) (Field, bool) {
 	return t.fields[i], true
 }
 
+type MapType struct {
+	value      *ListType
+	KeysSorted bool
+}
+
+func MapOf(key, item DataType) *MapType {
+	if key == nil || item == nil {
+		panic("arrow: nil key or item type for MapType")
+	}
+
+	return &MapType{value: ListOf(StructOf(Field{Name: "key", Type: key}, Field{Name: "value", Type: item, Nullable: true}))}
```

Comment:
does "key" and "value" hardcoded here have any implications for reading maps that name this differentlcy.

Reply:
I'll double check but I don't believe I enforce these names having to be named this way during reading. That said, naming them this way is how the spec describes it should be done.

Reply:
Ok, I confirmed that the names hardcoded here do not affect reading maps that name it differently. The assumption, as per the spec, is always that the first field is the keys and the second field is the values.

Reply:
+1, thanks.